lighthouse-core/gather/gather-runner.js

```diff
@@ -270,6 +278,12 @@ class GatherRunner {
           });
         });
         return artifacts;
+      })
+      // cleanup on error
+      .catch(err => {
+        GatherRunner.disposeDriver(driver);
```

Comment:
that way you can have an error specific log here. Even as simple as `log.log('status', 'Disconnecting from browser after error...'); or something